AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/CascadeDeleteOperation.swift

```diff
@@ -142,8 +146,19 @@ public class CascadeDeleteOperation<M: Model>: AsynchronousOperation {
         
         let modelIds = queriedModels.map { $0.identifier(schema: self.modelSchema).stringValue }
         
-        associatedModels = await self.recurseQueryAssociatedModels(modelSchema: self.modelSchema, ids: modelIds)
+        if modelName != "MutationEvent" {
+            self.log.debug("[CascadeDelete.1] Deleting \(modelName) with identifiers: \(modelIds)")
+        }
         
+        associatedModels = await self.recurseQueryAssociatedModels(modelSchema: self.modelSchema, ids: modelIds)
+        if self.log.logLevel >= .debug {
+            if modelName != "MutationEvent" {
+                self.log.debug("[CascadeDelete.2] Queried for \(modelName) associated models, found \(associatedModels.compactMap { $1.modelName })")
```

Comment:
doing this outside of the method recurseQueryAssociatedModels seems inefficient because we doing additional processing just for logging. I would look at recurseQueryAssociatedModels again to see if we have all the information available there to log. let's review recurseQueryAssociatedModels to see if we have all the information there

Reply:
tried doing this in `recurseQueryAssociatedModels` in couple of ways but looked more efficient and apt in the queryResult success case of `collapseResults` function.

Reply:
can you expand on your comment? i would like see the different ways you mentioned to better understand why this option is better